src/metabase/channel/render/js/svg.clj

```diff
@@ -42,9 +44,13 @@
     (js.engine/load-resource interface-path)))
 
 (def ^:private static-viz-context-delay
-  "Delay containing a graal js context. It has the chart bundle and the above `src-api` in its environment suitable
-  for creating charts."
-  (delay (load-viz-bundle (js.engine/context))))
+  "Delay containing a graal js context. It has the chart bundle and the above `src-api` in its environment suitable for
+  creating charts. However, under some circumstances, the Truffle JS engine tends to leak memory, so we don't want to
+  keep the reference to the engine forever. Ideally, we would like to make the engine lifetime bound to a single
+  request or something like that, but that would introduce a lot of changes and noise to the system. Hence, with this
+  bandaid, we keep the JS engine for some reasonable time (1 hour) before recreating it."
+  (delay/delay-with-ttl (.toMillis TimeUnit/HOURS 1)
```

Comment:
I would set the timeout as low as possible while keeping the overhead negligible. If instantiating the engine every 5-10 minutes has negligible overhead, then why don't we do that?